python/pyspark/sql/catalog.py

```diff
@@ -133,6 +133,17 @@ def listColumns(self, tableName, dbName=None):
                 isBucket=jcolumn.isBucket()))
         return columns
 
+    @since(3.2)
+    def tableExists(self, tableName, dbName=None):
+        """Check if the table or view with the specified name exists.
```

Comment:
I think you can remove `@since(3.2)` and add `.. versionadded:: 2.0.0` instead because `@since` doesn't work well with NumPy documentation style.

I think we should have a proper doc for each parameter and return value. (e.g., see https://spark.apache.org/docs/latest/api/python/_modules/pyspark/sql/dataframe.html#DataFrame.head)

Reply:
BTW, let's make the version as `3.3`. Branch for 3.2 is out, and now it's QA-period for 3.2 when we usually don't add new features.

Reply:
(don't forgot to add examples for doctests)

Reply:
Thanks for the comments. All implemented in the latest commit.